drivers/SmartThings/zigbee-lock/src/init.lua

```diff
@@ -57,6 +57,14 @@ local refresh = function(driver, device, cmd)
   device:refresh()
   device:send(LockCluster.attributes.LockState:read(device))
   device:send(Alarm.attributes.AlarmCount:read(device))
+  -- we can't determine from fingerprints if devices support lock codes, so
+  -- here in the driver we'll do a check once to see if the device responds here
+  -- and if it does, we'll switch it to a profile with lock codes
+  if not device:supports_capability_by_id(LockCodes.ID) and not device:get_field(lock_utils.CHECKED_CODE_SUPPORT) then
+    device:send(LockCluster.attributes.NumberOfPINUsersSupported:read(device))
+    -- we won't make this value persist because it's not that important
+    device:set_field(lock_utils.CHECKED_CODE_SUPPORT, true)
```

Comment:
This does mean that for devices that truly don't support lock codes, this will be re-checked every time the driver restarts.  I don't think that's a big deal, but wanted to call it out.

Reply:
I was _attempting_ to point that out with my comment, but I guess it wasn't that clear.